### PR TITLE
QPID-8482:[Broker-J]Derived a UUID for session name based on attach name

### PR DESCRIPTION
**Pull request by Dedeepya-T**
Derived a UUID for session name based on attach name and it can be used to log the session name.

### CODE REVIEW

File: broker-plugins/amqp-0-10-protocol/src/main/java/org/apache/qpid/server/protocol/v0_10/ServerConnectionDelegate.java [hunk 1]
```diff
@@ -427,6 +432,54 @@ public void sessionAttach(final ServerConnection serverConnection, final Session
...
+            hasher = MessageDigest.getInstance(MESSAGE_DIGEST_SHA256);
+            // Split the hash into two parts: MSB and LSB
+            long msb = toNumber(attachName, 0, 8);
+            long lsb = toNumber(attachName, 8, 16);
+
+            // Apply version and variant bits (required for RFC-4122 compliance)
+            msb = (msb & 0xffffffffffff0fffL) | (5 & 0x0f) << 12;
+            lsb = (lsb & 0x3fffffffffffffffL) | 0x8000000000000000L;
+
+            peerSessionName = new UUID(msb, lsb).toString();
```
Comment (alex-rufous) [inline broker-plugins/amqp-0-10-protocol/src/main/java/org/apache/qpid/server/protocol/v0_10/ServerConnectionDelegate.java:464]: Let's return Base64 representation of the digest bytes. I think that giving UUID might create an impression that the UUID was set on the client side whilst in fact it wasn't.

**Review by alex-rufous — CHANGES_REQUESTED** [review]
As per my in-line comments, let's have peer session name implemented only for AMQP 0-10 session.  Neither 0-9 nor 1-0 protocol have the remote session name concepts. Thus, the peer session name only make sense for 0-10 protocol. We can implement a management interface for 0-10 ( 'org.apache.qpid.server.model.Session')session extending 'org.apache.qpid.server.protocol.v0_10.Session'. The derived attribute peerSessionName can be declared on the 0-10 specific session interface and implemented in 0-10 session implementation.

File: broker-core/src/test/java/org/apache/qpid/server/session/AbstractAMQPSessionTest.java [hunk 2]
```diff
@@ -136,7 +136,7 @@ public void testRegisterTransactedMessageReceived()
 
         protected MockAMQPSession(final Connection parent, final int sessionId)
         {
-            super(parent, sessionId);
+            super(parent, sessionId,"mockAMQPSession");
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply broker-core/src/test/java/org/apache/qpid/server/session/AbstractAMQPSessionTest.java:139]: Done!

File: broker-core/src/main/java/org/apache/qpid/server/session/AMQPSession.java [hunk 3]
```diff
@@ -42,6 +42,9 @@
     @Override
     int getChannelId();
 
+    @Override
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/session/AMQPSession.java:45]: Removed!

File: broker-core/src/main/java/org/apache/qpid/server/model/Session.java [hunk 4]
```diff
@@ -51,6 +51,9 @@
     @DerivedAttribute
     int getChannelId();
 
+    @DerivedAttribute
+    String getPeerSessionName();
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/model/Session.java:55]: Moved it to the 0-10 session interface

File: broker-core/src/main/java/org/apache/qpid/server/session/AbstractAMQPSession.java [hunk 5]
```diff
@@ -152,6 +155,11 @@ public int getChannelId()
         return _sessionId;
     }
 
+    @Override
+    public String getPeerSessionName(){
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/session/AbstractAMQPSession.java:159]: Removed the method!

File: broker-plugins/amqp-0-10-protocol/src/main/java/org/apache/qpid/server/protocol/v0_10/ServerConnectionDelegate.java [hunk 6]
```diff
@@ -409,8 +413,9 @@ public void sessionAttach(final ServerConnection serverConnection, final Session
 
             final ServerSession serverSession =
                     new ServerSession(serverConnection, serverSessionDelegate, new Binary(atc.getName()), 0);
+            String peerSessionName = getPeerSessionName(serverSession.getName().getBytes());
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply broker-plugins/amqp-0-10-protocol/src/main/java/org/apache/qpid/server/protocol/v0_10/ServerConnectionDelegate.java:416]: Inlined!

File: broker-plugins/amqp-0-10-protocol/src/main/java/org/apache/qpid/server/protocol/v0_10/ServerConnectionDelegate.java [hunk 7]
```diff
@@ -427,6 +432,54 @@ public void sessionAttach(final ServerConnection serverConnection, final Session
         }
     }
 
+    private String getPeerSessionName(final byte[] attachName)
+    {
+        String peerSessionName = "";
+        try
+        {
+            peerSessionName = UUID.nameUUIDFromBytes(attachName).toString();
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply broker-plugins/amqp-0-10-protocol/src/main/java/org/apache/qpid/server/protocol/v0_10/ServerConnectionDelegate.java:440]: Done!

File: broker-plugins/amqp-0-10-protocol/src/main/java/org/apache/qpid/server/protocol/v0_10/ServerConnectionDelegate.java (hunk 1 above)
Comment (Dedeepya-T) [reply broker-plugins/amqp-0-10-protocol/src/main/java/org/apache/qpid/server/protocol/v0_10/ServerConnectionDelegate.java:464]: Done!

**Review by alex-rufous — APPROVED** [review]
The changes look reasonable for me

**Pull request closed without merging**